store/src/db/mod.rs

```diff
@@ -44,16 +47,25 @@ pub struct NoteCreated {
     pub note_index: u32,
     pub note_id: RpoDigest,
     pub sender: AccountId,
-    pub tag: u64,
+    pub tag: u32,
     pub details: Option<Vec<u8>>,
 }
 
 impl NoteCreated {
     /// Returns the absolute position on the note tree based on the batch index
-    /// and local-to-the-subtree index
+    /// and local-to-the-subtree index.
     pub fn absolute_note_index(&self) -> u32 {
         BlockNoteTree::note_index(self.batch_index as usize, self.note_index as usize) as u32
     }
+
+    // Returns the note type based on available details.
+    pub fn note_type(&self) -> NoteType {
+        if self.details.is_some() {
+            NoteType::Public
+        } else {
+            NoteType::OffChain
+        }
+    }
```

Comment:
This is probably OK for now, but note type should be a part of `NoteCreated` message (we'll do this in a different PR).

Reply:
Yeah, was counting on that as it seemed a bit more involved and depending on the leftover refactor for removing duplicate data.